### PR TITLE
Fix scaling bug

### PR DESCRIPTION
**Pull request by wrefgtzweve**
The bug was caused because i used ScrH( ) and ScrW( ) which won't work with other resolutions, stupid mistake.

### CODE REVIEW

**Comment by wrefgtzweve** [comment]
image
tested on cfc live, works fine

File: lua/cfc_loadouts/client/cl_cfc_loadout.lua [hunk 1]
```diff
@@ -288,10 +288,10 @@ function CFCLoadouts.openLoadout( )
     end
 
     local scrollDock = vgui.Create( "DScrollPanel", panel2 )
-    scrollDock:SetPos( ScrW( ) * 0.0825, ScrH( ) * 0.01 )
-    scrollDock:SetSize( ScrW( ) * 0.325, ScrH( ) * 0.49 )
+    scrollDock:SetPos( 158.4, 10.8 )
+    scrollDock:SetSize( 624, 529.2 )
     local weaponCats = vgui.Create( "DListLayout", scrollDock )
-    weaponCats:SetSize( ScrW( ) * 0.317, ScrH( ) * 0.4875 )
+    weaponCats:SetSize( 608.64,	526.5 )
```
Comment (samuelWilliams99) [inline lua/cfc_loadouts/client/cl_cfc_loadout.lua:294]:
```suggestion
    weaponCats:SetSize( 608.64, 526.5 )
```

File: lua/cfc_loadouts/client/cl_cfc_loadout.lua [hunk 2]
```diff
@@ -226,42 +226,42 @@ function CFCLoadouts.openLoadout( )
     end
 
     local saveLoadoutButton = vgui.Create( "DButton", panel2 )
-    saveLoadoutButton:SetPos( ScrW( ) * 0.005, ScrH( ) * 0.4125 )
-    saveLoadoutButton:SetSize( ScrW( ) * 0.075, ScrH( ) * 0.02 )
+    saveLoadoutButton:SetPos( 9.6, 445.5 )
+    saveLoadoutButton:SetSize( 144,	21.6 )
```
Comment (samuelWilliams99) [inline lua/cfc_loadouts/client/cl_cfc_loadout.lua:230]:
```suggestion
    saveLoadoutButton:SetSize( 144, 21.6 )
```

File: lua/cfc_loadouts/client/cl_cfc_loadout.lua [hunk 3]
```diff
@@ -148,38 +148,38 @@ function CFCLoadouts.openLoadout( )
...
 
     -----------------------
     -- Panel 2 panel2   ---
     -----------------------
     local weaponIcons = { }
     local loadoutListEditor = vgui.Create( "DListView", panel2 )
-    loadoutListEditor:SetPos( ScrW( ) * 0.005, ScrH( ) * 0.01 )
-    loadoutListEditor:SetSize( ScrW( ) * 0.075, ScrH( ) * 0.4 )
+    loadoutListEditor:SetPos( 9.6, 10.8 )
+    loadoutListEditor:SetSize( 144,	432 )
```
Comment (samuelWilliams99) [inline lua/cfc_loadouts/client/cl_cfc_loadout.lua:182]:
```suggestion
    loadoutListEditor:SetSize( 144, 432 )
```

File: lua/cfc_loadouts/client/cl_cfc_loadout.lua [hunk 4]
```diff
@@ -148,38 +148,38 @@ function CFCLoadouts.openLoadout( )
...
         net.WriteTable( selectedWeapons )
-        net.SendToServer( )
+        net.SendToServer()
     end
 
     local resetSelectButton = vgui.Create( "DButton", panel1 )
-    resetSelectButton:SetPos( ScrW( ) * 0.005, ScrH( ) * 0.4725 )
-    resetSelectButton:SetSize( ScrW( ) * 0.075, ScrH( ) * 0.025 )
+    resetSelectButton:SetPos( 9.6, 510.3 )
+    resetSelectButton:SetSize( 144,	27 )
```
Comment (samuelWilliams99) [inline lua/cfc_loadouts/client/cl_cfc_loadout.lua:167]:
```suggestion
    resetSelectButton:SetSize( 144, 27 )
```

File: lua/cfc_loadouts/client/cl_cfc_loadout.lua [hunk 5]
```diff
@@ -95,26 +95,26 @@ function CFCLoadouts.openLoadout( )
...
-    weaponLoadoutPreviewScroll:SetSize( ScrW( ) * 0.325, ScrH( ) * 0.5 )
-    weaponLoadoutPreviewScroll:Hide( )
+    weaponLoadoutPreviewScroll:SetPos( 158.4,	10.8 )
+    weaponLoadoutPreviewScroll:SetSize( 624, 540 )
+    weaponLoadoutPreviewScroll:Hide()
     local loadoutPreviewList = vgui.Create( "DListView", panel1 )
-    loadoutPreviewList:SetPos( ScrW( ) * 0.005, ScrH( ) * 0.01 )
-    loadoutPreviewList:SetSize( ScrW( ) * 0.075, ScrH( ) * 0.4325 )
+    loadoutPreviewList:SetPos( 9.6,	10.8 )
+    loadoutPreviewList:SetSize( 144,	467.1 )
```
Comment (samuelWilliams99) [inline lua/cfc_loadouts/client/cl_cfc_loadout.lua:111]:
```suggestion
    loadoutPreviewList:SetPos( 9.6, 10.8 )
    loadoutPreviewList:SetSize( 144, 467.1 )
```

**Review by samuelWilliams99 — CHANGES_REQUESTED** [review]
Does having these hard coded mean it'll be tiny for higher res monitors?
It might be worth calculating a scaling factor, and using that for everything
That way the menu doesnt ever change shape/ratio, but can change size.

The factor could be as simple as the screen height

**Comment by wrefgtzweve** [comment]
> Does having these hard coded mean it'll be tiny for higher res monitors?
> It might be worth calculating a scaling factor, and using that for everything
> That way the menu doesnt ever change shape/ratio, but can change size.
> 
> The factor could be as simple as the screen height

Eh, it's 800x600. Yes it might be but in it's current state it doesn't work at all and i'd have no idea / don't feel like looking into how to make it scale, this entire project made me fucking hate vgui

**Comment by samuelWilliams99** [comment]
What resolution did u dev this on?

**Comment by wrefgtzweve** [comment]
1920x1080

**Comment by samuelWilliams99** [comment]
Mind if I just commit the changes for scaling on here?

**Comment by wrefgtzweve** [comment]
feel free to!

**Comment by samuelWilliams99** [comment]
Would you be able to test the changes with a few different res'?

**Comment by wrefgtzweve** [comment]
will do

**Review by samuelWilliams99 — APPROVED** [review]

**Comment by wrefgtzweve** [comment]
fyi, i can't merge due to jenkins not working

**Pull request merged**